src/wallet/wallet.cpp

```diff
@@ -2442,7 +2442,11 @@ void CWallet::AvailableCoins(interfaces::Chain::Lock& locked_chain, std::vector<
             continue;
         }
 
-        if (nDepth < nMinDepth || nDepth > nMaxDepth)
+        int min_depth = 0;
+        if (coinControl)
+            min_depth = coinControl->m_min_depth;
+
+        if (nDepth < min_depth || nDepth > nMaxDepth)
```

Comment:
style-nit: Add those `{}`

```suggestion
        if (nDepth < min_depth || nDepth > nMaxDepth) {
```